device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsTransport.java

```diff
@@ -55,6 +55,9 @@
     /** Messages whose callbacks that are waiting to be invoked. */
     private final Queue<IotHubCallbackPacket> callbackList = new LinkedBlockingDeque<>();
 
+    /** Connection state change callbacks */
+    private final List<IotHubConnectionStateCallback> stateCallbackList = new ArrayList<>();
+
```

Comment:
Why would a list of callbacks be needed ? We would only need one per SDK. May be just having a member variable storing callback and context would suffice here. ArrayList won't be needed (unless you had a scenario in mind that I cannot see right now. Please provide details if so.)